### PR TITLE
Feature/reorg lib

### PR DESCRIPTION
**Pull request by astro019**

### CODE REVIEW

File: cep47/src/cep47.rs [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        Owners::instance().get(&token_id)
+    }
+
+    fn token_meta(&self, token_id: TokenId) -> Option<Meta> {
+        Metadata::instance().get(&token_id)
+    }
+
+    fn set_token_meta(&mut self, token_id: TokenId, meta: Meta) -> Result<(), Error> {
+        if self.owner_of(token_id.clone()).is_none() {
+            return Err(Error::TokenIdAlreadyExists);
```
Comment (deuszex) [inline cep47/src/cep47.rs:68]: Shouldn't this return `TokenIdDoesntExist`?

**Review by deuszex — COMMENTED** [review]
Description about the purpose and details of the PR would be appreciated.
Didn't find tests regarding the re-implementation of "tokens" dictionary handling.

**Comment by astro019** [comment]
We are not able to track the tokenIds in current CEP47 master.
For example, when an item is minted without custom tokenId, we don't know the tokenId and cannot transfer the item.
In the other branches of CEP47 - feature/get_caller, there is "tokens()" endpoint to get tokenId lists that each account holds. However, increasing the amount of tokens that an account owns causes linear increase of gas used. Thus this endpoint is removed in PR - https://github.com/casper-ecosystem/casper-nft-cep47/pull/20.

In current PR, we provide "get_token_by_index" endpoint to allow a party to get tokenId that it owns with index.
In details, it has "owned_tokens_by_index" named_key to get tokenId with index and "owned_indexes_by_token" named_key to get index with tokenId.

**Comment by devendran-m** [comment]
As per @deuszex 's, 

> This pr was superseeded by id_increment which was merged in january. I think the source branch was closed too.

Therefore closing this PR

**Pull request closed without merging**